src/model/TickDataSource.ts

```diff
@@ -71,11 +74,39 @@ export default class TickDataSource extends MongoDataSource<TickType> {
     }
   }
 
-  async ticksByUser (userId: string): Promise<TickType[]> {
-    return await this.tickModel.find({ userId })
+  /**
+   * Retrieve ticks of a user given their details
+   * @param userSelectors Attributes that can be used to identify the user
+   * @returns
+   */
+  async ticksByUser (userSelectors: TickUserSelectors): Promise<TickType[]> {
+    const { userId: requestedUserId, username } = userSelectors
+    if (requestedUserId == null && username == null) {
+      throw new Error('Username or userId must be supplied')
+    }
+    const filters: any[] = []
+    if (requestedUserId != null) {
+      filters.push({ _id: requestedUserId })
+    }
+    if (username != null) {
+      filters.push({ // Fuzzy match based on canonicalName since those are unique to a user.
```

Comment:
The other reason I thought we should filter using `canonicalName` is because we have an index on it but not on `username`.

Reply:
Not having an index on username was an oversight since it's an important query for the profile page. 
The 'fuzzy' match was only intended for checking username uniqueness when updating/creating username.

To simplify things I don't think we should support fetching ticks by canonical name.  What do you think?
We can look into supporting a true typo-tolerant username matching in the future if there's a use case for it.

Reply:
Sounds good. I can fix it so that it searches strictly by username when I rebase on top of #312 !